pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -490,6 +493,23 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusterForPortOrUDS(pluginPara
 	localityLbEndpoints := buildInboundLocalityLbEndpoints(pluginParams.Bind, instance.Endpoint.EndpointPort)
 	localCluster := buildDefaultCluster(pluginParams.Push, clusterName, apiv2.Cluster_STATIC, localityLbEndpoints,
 		model.TrafficDirectionInbound, pluginParams.Node, nil, false)
+	endpointIP := net.ParseIP(pluginParams.Bind)
+	if endpointIP != nil {
+		localAddr := LocalhostIPv6Address
+		if endpointIP.To4() != nil {
+			localAddr = LocalhostAddress
+		}
+		// specific the bind address for inbound traffic redirect to local service.
+		// this configuration is used to distinguish traffic between inbound traffic to specific podIP and outbound traffic back to itself using VIP.
+		localCluster.UpstreamBindConfig = &core.BindConfig{
+			SourceAddress: &core.SocketAddress{
+				Address: localAddr,
```

Comment:
I am not sure  it is the safe way to do it. @costinm

Reply:
We have already used this feature to bind `127.0.0.6` for `InboundPassthroughClusterIpv4` cluster

Reply:
Please donot set local host for all the cases, we may support preserving src ip address in future.

Reply:
I think it is safe to do this when bind address is not localhost